modules/cudafeatures2d/src/cuda/orb.cu

```diff
@@ -56,13 +58,17 @@ namespace cv { namespace cuda { namespace device
         ////////////////////////////////////////////////////////////////////////////////////////////////////////
         // cull
 
-        int cull_gpu(int* loc, float* response, int size, int n_points)
+        int cull_gpu(int* loc, float* response, int size, int n_points, cudaStream_t stream)
         {
             thrust::device_ptr<int> loc_ptr(loc);
             thrust::device_ptr<float> response_ptr(response);
-
-            thrust::sort_by_key(response_ptr, response_ptr + size, loc_ptr, thrust::greater<float>());
-
+            if(stream)
+            {
+                thrust::sort_by_key(thrust::cuda::par.on(stream), response_ptr, response_ptr + size, loc_ptr, thrust::greater<float>());
```

Comment:
AFAIK, this will work only with CUDA 7.0 and newer. Please increase minimal required CUDA version in `opencv/cmake/OpenCVMinDepVersions.cmake` and in `opencv/modules/core/include/opencv2/core/private.cuda.hpp`.